sdk/inc/azure/storage/az_storage_blobs.h

```diff
@@ -108,39 +108,39 @@ typedef struct
 AZ_NODISCARD az_storage_blobs_blob_client_options az_storage_blobs_blob_client_options_default();
 
 /**
- * @brief Gets the default blob upload options
+ * @brief Gets the default blob upload options.
  *
- * @details Call this to obtain an initialized #az_storage_blobs_blob_upload_options structure
+ * @details Call this to obtain an initialized #az_storage_blobs_blob_upload_options structure.
  *
  * @remark Use this, for instance, when only caring about setting one option by calling this method
  * and then overriding that specific option.
- *
  */
 AZ_NODISCARD AZ_INLINE az_storage_blobs_blob_upload_options
 az_storage_blobs_blob_upload_options_default()
 {
   return (az_storage_blobs_blob_upload_options){ .context = &az_context_application,
-                                                ._internal = { .unused = AZ_SPAN_NULL } };
+                                                 ._internal = { .unused = AZ_SPAN_NULL } };
 }
 
 /**
  * @brief Uploads the contents to blob storage.
  *
- * @param client A storage blobs client structure.
- * @param content The blob content to upload.
- * @param options __[nullable]__ A reference to an #az_storage_blobs_blob_upload_options
+ * @param[in] client An #az_storage_blobs_blob_client structure.
+ * @param[in] content The blob content to upload.
+ * @param[in] options __[nullable]__ A reference to an #az_storage_blobs_blob_upload_options
  * structure which defines custom behavior for uploading the blob. If `NULL` is passed, the client
- * will use the default options (i.e. #az_storage_blobs_blob_upload_options_default()).
- * @param response A pre-allocated buffer where to write HTTP response into.
+ * will use the default options (i.e. #az_storage_blobs_blob_upload_options_default).
+ * @param[in,out] ref_response An initialized `az_http_response` where to write HTTP response into.
  *
- * @return An #az_result value indicating the result of the operation:
- *         - #AZ_OK if successful
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK Success.
+ * @retval other Failure.
  */
 AZ_NODISCARD az_result az_storage_blobs_blob_upload(
     az_storage_blobs_blob_client* client,
     az_span content, /* Buffer of content*/
-    az_storage_blobs_blob_upload_options* options,
-    az_http_response* response);
+    az_storage_blobs_blob_upload_options const* options,
+    az_http_response* ref_response);
```

Comment:
Shouldn't this be `out_response`?

Reply:
From offline discussion with @antkmsft, no, it should remain ref. We should document this in storage.